src/tiled/objectiddialog.cpp

```diff
@@ -53,8 +53,8 @@ ObjectIdDialog::ObjectIdDialog(QWidget *parent)
 
     Utils::restoreGeometry(this);
 
-    if (document = DocumentManager::instance()->currentDocument()) {
-        if (map = document->map()) {
+    if ((document = DocumentManager::instance()->currentDocument())) {
+        if ((map = document->map())) {
```

Comment:
I'd prefer if you fixed these warnings by instead localizing the scope of those variables. There won't be a warning when you write it like:
```c++
if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
    if (Map *map = document->map()) {
```
In addition, note that there is never a need for that second check. A `MapDocument` _always_ has a `Map` instance.